lib/trino-filesystem-azure/src/main/java/io/trino/filesystem/azure/AzureLocation.java

```diff
@@ -41,30 +41,38 @@ private AzureLocation(Location location)
 
         // container is interpolated into the URL path, so perform extra checks
         location.userInfo().ifPresent(container -> {
-            checkArgument(!container.isEmpty(), INVALID_LOCATION_MESSAGE, this.location);
+            checkArgument(!container.isEmpty(), INVALID_LOCATION_MESSAGE, location);
             checkArgument(
                     CONTAINER_VALID_CHARACTERS.matchesAllOf(container),
-                    "Invalid Azure storage account name. Valid characters are 'a-z', '0-9', and '-': %s",
-                    this.location);
+                    "Invalid Azure storage container name. Valid characters are 'a-z', '0-9', and '-': %s",
+                    location);
+            checkArgument(
+                    !container.startsWith("-") && !container.endsWith("-"),
+                    "Invalid Azure storage container name. Cannot start or end with a hyphen: %s",
+                    location);
             checkArgument(
                     !container.contains("--"),
-                    "Invalid Azure storage account name. Cannot contain double dash: %s",
-                    this.location);
+                    "Invalid Azure storage container name. Cannot contain consecutive hyphens: %s",
+                    location);
         });
 
         // storage account is the first label of the host
-        checkArgument(location.host().isPresent(), INVALID_LOCATION_MESSAGE, this.location);
+        checkArgument(location.host().isPresent(), INVALID_LOCATION_MESSAGE, location);
         String host = location.host().get();
         int accountSplit = host.indexOf('.');
         checkArgument(
                 accountSplit > 0,
                 INVALID_LOCATION_MESSAGE,
                 this.location);
         this.account = host.substring(0, accountSplit);
+
+        // host must end with ".dfs.core.windows.net"
+        checkArgument(host.substring(accountSplit).equals(".dfs.core.windows.net"), INVALID_LOCATION_MESSAGE, location);
```

Comment:
I think this needs to use the `accountSplit` index. Otherwise, this would succeed for a host with extra parts like `abc.xyz.dfs.core.windows.net`

Reply:
This is using `accountSplit`, and there is a test for this case